### ISSUE TITLE
Any plans for SPM support?

### ISSUE THREAD

**Issue by RolandasRazma** [bugfender/BugfenderSDK-iOS#57]
Any plans for SPM support?

**Comment by jgimenez** [comment]
SPM currently does not support binary frameworks. We plan to add support with Swift 5.3: https://github.com/apple/swift-evolution/blob/master/proposals/0272-swiftpm-binary-dependencies.md

**Comment by acarrera94** [comment]
Looks like swift 5.3 is out with Xcode 12. What does does your time line for SPM support look like?

**Comment by RubenVot** [comment]
Thanks a lot @acarrera94 !! We are already working on this and it should be ready in a week or two.

**Comment by acarrera94** [comment]
Great! excited to add it to my app!

**Comment by RubenVot** [comment]
We are starting to test the SPM distribution in our beta branch
https://github.com/bugfender/BugfenderSDK-iOS/tree/beta

Feel free to download and notify any error you might find :)

**Comment by acarrera94** [comment]
Thanks for doing that! 
So when I add the package and import it into my project it crashes with `signal SIGABRT` and I get this in the logs: 
```
dyld: Library not loaded: @rpath/BugfenderSDK.framework/BugfenderSDK
  Referenced from: /private/var/containers/Bundle/Application/CD93102A-E312-48F5-908D-E0E769B080ED/Droplii 2.0.app/Droplii 2.0
  Reason: image not found
dyld: launch, loading dependent libraries
DYLD_LIBRARY_PATH=/usr/lib/system/introspection
DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
```

**Comment by RubenVot** [comment]
Hi @acarrera94 thanks a lot for providing feedback! 
Which version of Xcode are you using (to try to reproduce in our side)

**Comment by acarrera94** [comment]
Xcode 12 beta 2. It even happens when creating a brand new iOS SwiftUI app
and just adding the beta branch of bugfender

**Comment by RubenVot** [comment]
Hi @acarrera94 sorry, I missed one step in the documentation. 
You have to drag the framework manually. At some point this should be done by the SPM but I can't find docs on how to automate it.
Just drag and drop from the "Referenced Binaries" and it should work now.  

<img width="1043" alt="Screenshot 2020-08-14 at 14 11 41" src="https://user-images.githubusercontent.com/1536171/90248220-92c2db80-de38-11ea-979b-0b88983c604c.png">

**Comment by RubenVot** [comment]
Hi @acarrera94, we just pushed a new package update and the additional step shouldn't be necessary anymore. You can remove and add the dependency and everything should just work. 
We are leaving this change in the **beta** branch for a week and probably moving to master soon.

**Comment by lecniers** [comment]
@RubenVot I tried the beta release and didn't work for me out of the box, I had to use it as a local package (like @acarrera94 ) posted, but now I can't upload a new build to the store I'm getting this errors:

![Screen Shot 2020-09-30 at 2 18 26 PM](https://user-images.githubusercontent.com/6807647/94724203-ed03f700-0327-11eb-8cc1-f4f8ba545300.png)

**Comment by RolandasRazma** [comment]
this is SPM/Xcode bug. You can track it here https://bugs.swift.org/browse/SR-13343

**Comment by acarrera94** [comment]
I actually had it working before but now when adding the beta branch I get this: 
```
artifact of binary target 'BugfenderSDK' failed download: invalid status code 404
```

It started happening once I removed Bugfender from SPM for the project then added it again.

**Comment by acarrera94** [comment]
Looks like the url in the Package.swift is returning a 404. 
https://github.com/bugfender/BugfenderSDK-iOS/releases/download/1.9.3/BugfenderSDK.xcframework.zip

Any idea when we can expect that to be resolved?

**Comment by reden87** [comment]
The latest commit points to a release which is not yet published: https://github.com/bugfender/BugfenderSDK-iOS/commit/a5431af95fa7ec30cf9304ffed0c696a152aee6c 1.9.2 is the latest public release @RubenVot

**Comment by Gujci** [comment]
Sticking to commit `af1461c6ba0e1a8ac5d16cf7f8c55ae061d1449a` seems to work. The ones after that are possibly the preparation of release 1.9.3, which will have SPM in it, I assume.

**Comment by RubenVot** [comment]
Yes, you are right. 
Sorry for the delay in this release. We were planning to release a few weeks ago but we decided to stop because there was a bug in cocoapods that made the app unusable.  
They have finally fixed it yesterday. We need to run all the tests again and finally publish. 
Again, sorry for the inconvenience!

**Comment by RubenVot** [comment]
Pre-Released Bugfender 1.10.0 
Testing now.

**Comment by RubenVot** [comment]
Released

**Comment by acarrera94** [comment]
I'm seeing "Unable to Install App" when running debug on my device. Is this related to the issue mentioned earlier? The weird thing is that when I build for release it works just fine. And if I remove Bugfender dependency from the project it works as well. 

Happy to create a new issue if needed. 

Full Log:

```
Details

Unable to install "App"
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402620395
--
A valid provisioning profile for this executable was not found.
Domain: com.apple.dt.MobileDeviceErrorDomain
Code: -402620395
User Info: {
    DVTRadarComponentKey = 487925;
    MobileDeviceErrorCode = "(0xE8008015)";
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x000000012803b93f DTDKCreateNSErrorFromAMDErrorCode + 220
	1   DTDeviceKitBase                     0x000000012807a124 __90-[DTDKMobileDeviceToken installApplicationBundleAtPath:withOptions:andError:withCallback:]_block_invoke + 155
	2   DVTFoundation                       0x000000010b7b9b33 DVTInvokeWithStrongOwnership + 71
	3   DTDeviceKitBase                     0x0000000128079e65 -[DTDKMobileDeviceToken installApplicationBundleAtPath:withOptions:andError:withCallback:] + 1440
	4   IDEiOSSupportCore                   0x0000000127eead28 __118-[DVTiOSDevice(DVTiPhoneApplicationInstallation) processAppInstallSet:appUninstallSet:installOptions:completionBlock:]_block_invoke.292 + 3513
	5   DVTFoundation                       0x000000010b8e829a __DVT_CALLING_CLIENT_BLOCK__ + 7
	6   DVTFoundation                       0x000000010b8e9ebc __DVTDispatchAsync_block_invoke + 1191
	7   libdispatch.dylib                   0x00007fff6e3056c4 _dispatch_call_block_and_release + 12
	8   libdispatch.dylib                   0x00007fff6e306658 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x00007fff6e30bc44 _dispatch_lane_serial_drain + 597
	10  libdispatch.dylib                   0x00007fff6e30c5d6 _dispatch_lane_invoke + 363
	11  libdispatch.dylib                   0x00007fff6e315c09 _dispatch_workloop_worker_thread + 596
	12  libsystem_pthread.dylib             0x00007fff6e560a3d _pthread_wqthread + 290
	13  libsystem_pthread.dylib             0x00007fff6e55fb77 start_wqthread + 15
);
}
--


System Information

macOS Version 10.15.7 (Build 19H2)
Xcode 12.1 (17222)
```

**Comment by RubenVot** [comment]
Hi @acarrera94, just a note, we released yesterday Bugfender 1.10, yo can use now the master branch and remove the references to beta. Probably remove Bugfender and add it again replacing branch with last version in the SPM wizard. 

So, the issue you mention doesn't appear to be related to Bugfender. Looks like Xcode can't sign the app for your device. 
I just created a new sample project using Xcode 12.1, followed the instructions in [the readme](https://github.com/bugfender/BugfenderSDK-iOS/blob/master/README.md) and ran the project in an iPhoneX and an iPad Pro with no issues. 
Which device are you using? Can you confirm that the problem persists after delete the app from the device, switch off/switch on and running again? Finally I would recommend to go in Xcode to Window -> Devices & Simulators and double check there isn't any concrete error right there. 
If the problem persist, could you please open a new issue with a sample project so we can take a look? 

Thanks for your report, good luck debugging!
